core/templates/dev/head/components/question-directives/question-player/question-player.directive.ts

```diff
@@ -416,6 +514,32 @@ oppia.directive('questionPlayer', [
             openConceptCardModal(ctrl.failedSkillIds);
           };
 
+          ctrl.openSkillMasteryModal = function(skillId) {
+            if (ctrl.userIsLoggedIn) {
+              $uibModal.open({
+                templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+                  '/components/question-directives/question-player/' +
+                  'skill-mastery-modal.template.html'),
+                backdrop: true,
+                controller: [
+                  '$scope', '$uibModalInstance',
+                  function(
+                      $scope, $uibModalInstance) {
+                    $scope.skillId = skillId;
+                    $scope.masteryChange = ctrl.masteryPerSkillMapping[skillId];
+
+                    $scope.closeModal = function() {
+                      $uibModalInstance.dismiss('cancel');
+                    };
+                  }
+                ]
+              });
+            } else {
+              AlertsService.addWarning(
```

Comment:
Done.

Reply:
I agree with @vinitamurthi not to show a warning. 

I don't know if I'm that good at copywriting, but you could perhaps try: "Log in to Oppia to view your long-term skill mastery." perhaps. I think that's simple and straightforward, and should work well for now. Don't forget to internationalize this string since it is learner-facing.

Also, one question I have here is: what is the flow for that case? If the user clicks that button, and is redirected to the log in screen, will they then come back to this page to see their skill mastery details? It's a flow that's probably worth testing in e2e tests.

Reply:
No it won't redirect guests to the log in screen. It will just show a message reminding them this is a feature only for logged in users.

Reply:
Hm. Perhaps "log in" should be a link that does the same thing as what the login button on the top right does. If the user does log in following the advice of this modal, what happens after they do so? Which page do they land on after logging in?

Reply:
So, are you suggesting to redirect guests to the login page? If so, they will be redirected back to the review test result page again.

Reply:
And the existing review test result data is preserved?

If so, that sounds good. You don't need to redirect them actively, but give them the option to do so, with a link under "log in".

Reply:
Umm, I'm not so sure about preserving the result data yet. I'll look into it first.

Reply:
OK. If it's not preserved, then since this comes at the very end of a test (right?) perhaps it's OK to redirect them to the topic home page for practice sessions, or something like that, and it won't interrupt the middle of the experience.